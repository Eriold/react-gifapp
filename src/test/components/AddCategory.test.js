import React from 'react'
const { shallow } = require('enzyme')
const { AddCategory } = require('../../Components/AddCategory')
import '@testing-library/jest-dom'

describe('test component <AddCategory />', () => {

  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />)
  const value = 'Hello World'

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />)
  })

  test('show component', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('change input ', () => {
    const input = wrapper.find('input');
    input.simulate('change', { target: { value } });
    const inputValueText = wrapper.find('p').text().trim();
    expect(inputValueText).toBe(value)
  })

  test('do not send submit null', () => {
    wrapper.find('form').simulate('submit', { preventDefault() { } });
    expect(setCategories).not.toHaveBeenCalled();
  })

  test('call setCategories and clean input ', () => {
    wrapper.find('input').simulate('change', { target: { value } });
    wrapper.find('form').simulate('submit', { preventDefault() { } });
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(wrapper.find('input').prop('value')).toBe('')
  })

})