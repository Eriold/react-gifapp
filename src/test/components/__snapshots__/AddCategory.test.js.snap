// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component <AddCategory /> show component 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
